universe/interface.go

```diff
@@ -74,6 +81,55 @@ func (i *Identifier) StringForLog() string {
 		i.String(), i.AssetID[:], groupKey)
 }
 
+// Encode encodes an identifier as a byte array.
+func (i *Identifier) Encode(w io.Writer) error {
```

Comment:
Why this vs encoding directly in the SQL table? Ideal to not pin down the identifier in the schema itself?